go/vt/vtgate/tx_conn.go

```diff
@@ -81,11 +84,14 @@ func (txc *TxConn) commit2PC(ctx context.Context, session *SafeSession) error {
 		return err
 	}
 
-	err = txc.multiGo(session.ShardSessions[1:], func(s *vtgatepb.Session_ShardSession) error {
+	err = txc.multiGoSessions(session.ShardSessions[1:], func(s *vtgatepb.Session_ShardSession) error {
 		return txc.gateway.Prepare(ctx, s.Target, s.TransactionId, dtid)
 	})
 	if err != nil {
-		// TODO(sougou): We could call RollbackPrepared here once it's ready.
+		// TODO(sougou): Perform a more fine-grained cleanup
+		// including unprepared transactions.
+		_ = txc.Resume(ctx, dtid)
```

Comment:
Should we log this error, at least? Maybe throttled logger, although the throttling might be overkill...